commons/src/main/java/org/infinispan/commons/api/functional/Closeables.java

```diff
@@ -0,0 +1,26 @@
+package org.infinispan.commons.api.functional;
+
+import java.util.Spliterator;
+
+// TODO: Remove once Will's branch closeable spliterator is in.
+
+/**
+ * @since 8.0
+ */
+public class Closeables {
+
+   private Closeables() {
+      // Cannot be instantiated, it's just a holder class
+   }
+
+   /**
+    * Auto closeable spliterator. Being able to close an spliterator is useful
+    * for situations where the iterator might consume considerable resources,
+    * e.g. in a distributed environment. Hence, being able to close it allows
+    * for resources to be reclaimed if the iterator won't be iterated any more.
+    */
+   public interface CloseableSpliterator<T> extends Spliterator<T>, AutoCloseable {
```

Comment:
@galderz well, I'm not very familiar with the new collections stuff in Java 8, so I'm not sure what you *could* do with a `Spliterator` inside that block... I was actually expecting that you'd have reusable algorithms that work on any `Collection` and call `spliterator()` themselves, sort of like for-each calls `iterator()`. But the JDK doesn't have any examples of that, so perhaps the user is expected to call `spliterator()` explicitly after all. Besides, `Traversable` doesn't extend `Collection`, so that kind of generic algorithm wouldn't work anyway.

BTW, do you know a good resource for learning about the streams stuff in Java 8?